src/Component/src/Symfony/Routing/Factory/Resource/ResourceRouteCollectionFactoryInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Symfony\Routing\Factory\Resource;
+
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * @experimental
+ */
+interface ResourceRouteCollectionFactoryInterface
+{
+    /** @psalm-param class-string $className */
```

Comment:
```suggestion
    /** @param class-string $className */
```

I think it should be supported out of the box without analyzer specific docblock.
 
https://phpstan.org/writing-php-code/phpdoc-types#class-string